features/environment.py

```diff
@@ -192,12 +191,21 @@ def _make_patroni_test_config(self, name, custom_config):
             config['raft'] = {'data_dir': self._output_dir, 'self_addr': 'localhost:' + os.environ['RAFT_PORT']}
 
         host = config['restapi']['listen'].rsplit(':', 1)[0]
-        config['restapi']['listen'] = config['restapi']['connect_address'] = '{}:{}'.format(host, 8008 + int(name[-1]))
+
+        try:
+            port = 8008 + int(name[-1])
+        except ValueError:
+            port = 8008
+        config['restapi']['listen'] = config['restapi']['connect_address'] = '{}:{}'.format(host, port)
 
         host = config['postgresql']['listen'].rsplit(':', 1)[0]
         config['postgresql']['listen'] = config['postgresql']['connect_address'] = '{0}:{1}'.format(host, self.__PORT)
 
-        config['name'] = name
+        if custom_config and 'name' in custom_config:
+            config['name'] = custom_config['name']
+        else:
+            config['name'] = name
+        
```

Comment:
I believe if I make this change, in order to keep the integration tests working, I'd need to make a similar change elsewhere. Here's what I was thinking:

- What we want is for two nodes to have the same `name` in their configuration. However, in `environment.py` and `PatroniPoolController` we also use `name` as the unique key in the dictionary keeping track of processes.
- So the way I got around this was by having the `name` be `dup_postgres0` in the PoolController, but using this if statement to ensure that in the configuration for the node, `name` is `postgres0`

Making this change as is would make it so that the `name` parameter of the duplicate node would be `dup_postgres0` which wouldn't cause the issue. If I changed the way the duplicate node is spawned, then it would overwrite the original `postgres0` process in `pctl._processes`. My thought was that this if statement allowed me to track the different processes with different names, but ensure that in their configurations they did indeed have duplicate names.

What are your thoughts on this? Is there something I'm missing that would allow me to cut these lines or solve the same problem in a better way?

Reply:
https://github.com/zalando/patroni/blob/e72d3ba79e92708c378f48eec84979349dd44197/features/environment.py#L244-L245 already does everything we need.